lib/URL-impl.js

```diff
@@ -19,9 +20,12 @@ exports.implementation = class URLImpl {
       throw new TypeError("Invalid URL");
     }
 
+    const query = parsedURL.query || "";
+
     this._url = parsedURL;
 
-    // TODO: query stuff
+    this._query = URLSearchParams.createImpl([query], { doNotStripQMark: true });
```

Comment:
Yeah, OK, I guess this is tied up in how we have no way in webidl2js to create "a new URLSearchParams object" without invoking the constructor. Which is not great in and of itself. Maybe add a comment explaining what we are working around.